test/sharness/t0040-add-and-cat.sh

```diff
@@ -78,6 +78,56 @@ test_add_cat_file() {
     test_expect_code 1 ipfs cat --offset -102 "$HASH" > actual
   '
 
+  test_expect_success "ipfs cat with length succeeds" '
+    ipfs cat --length 8 "$HASH" >actual
+  '
+
+  test_expect_success "ipfs cat with length output looks good" '
+    echo -n "Hello Wo" >expected &&
+    test_cmp expected actual
+  '
+
+  test_expect_success "ipfs cat multiple hashes with offset and length succeeds" '
+    ipfs cat --offset 5 --length 15 "$HASH" "$HASH" "$HASH" >actual
+  '
+
+  test_expect_success "ipfs cat multiple hashes with offset and length looks good" '
+    echo " Worlds!" >expected &&
+    echo -n "Hello " >>expected &&
```

Comment:
I would use `printf "Worlds!\nHello '"`.

Reply:
Note that is you try this in bash on the command line this will fail as '!' has special meaning in bash when history expansion in enabled.  This is disabled by default in scripts so it won't cause a problem.